docs/designs/high-level.md

```diff
@@ -1,33 +1,104 @@
 
-# Exporters
-1. Export metrics and OS knowledge.
+See [Custom Resource Reference Guide](crd.md) for a full reference to the custom resource.
 
+### Architecture Diagram
 
-Most interactions with the operator will be performed through the OpenSearch operator.
-This can be done in 2 ways:
+```mermaid
+  flowchart TD
+      subgraph operator [OpenSearch Operator]
+          direction LR
+          controller[Controller] -->manager
+      end
+      subgraph manager [Manager]
+          direction TB
+          subgraph reconcilers [Reconcilers Chain]
+              reconciler1[Reconciler 1] --- reconciler2
+              reconciler2[Reconciler 2] --- reconciler3
+              reconciler3[Reconciler 3] --- reconciler_etc[...]
+          end
+          subgraph components
+              scaler([Scaler component])
+              auto-scaler([Auto-Scaler Component]) 
+          end
+      end
+      
+      controller-- watches -->cr{{OpenSearchCluster Custom Resource}}
+      
+      subgraph kubernetes-objects-created [Created K8s Objects]
+          direction LR
+          opensearch-cluster-statefulsets[OpenSearch Cluster Statefulsets]
+          openseach-config-map[OpenSearch ConfigMap]
+          opensearch-service[OpenSearch Service]
+          ...
+      end    
+
+      subgraph monitoring [Monitoring Tools]
+          prometheus[Prometheus]
+          grafana[Grafana]
+          alert-manager[Alert Manager]
+      end
+      
+      manager --> kubernetes-objects-created
+```
+
+### Provisioning 
+The operator contains a Controller for its Custom Resource (`OpenSearchCluster`). Following the Controller pattern, it is mainly an endless loop (a.k.a. [Control Loop](https://en.wikipedia.org/wiki/Control_loop)), waiting for changes made to the custom resources, and responds to the changes by adjusting the deployed cluster to match the desired configuration specified in the custom resource. It can be a simple change to the Security configuration which may only update dynamic settings using OpenSearch API,or it can also be a change to the pod spec, increasing memory requirements from 10gb to 12gb. The latter requires a series of steps to accomplish that.
+
+Once a change has been detected it activates a Manager with the changed custom resource ID. The Manager goal is to modify/create/delete the OpenSearch Cluster and other deployed tools (e.g. Prometheus) reflecting the changes made to the custom resource. 
+
+The Manager is primarily composed of two types of elements: Reconcilers and Components. The Reconcilers main job is to update/create the deployed Kubernetes objects representing the OpenSearch Cluster and other various software the Operator needs. If we take the Custom Resource YAML and split into it logical parts (i.e. fields), then a Reconciler is mapped to a specific part (field). The operators have the following Reconcilers:
```

Comment:
From the naming I think this should be the Controller instead of Manager. My understanding is that the manager does some orchestration in the background but the code that actually deals with a custom object is the Controller which in our case delegates the actual work to the Reconcilers.

Reply:
@idanl21 ⬆️